src/connections/relatedRelationships.js

```diff
@@ -0,0 +1,263 @@
+  newRelatedRel.forEach(newRel => {
+    const oldRel = oldRelatedRel.find(oldRel => oldRel._id === newRel._id)
+    if (!oldRel) {
+      relatedRelToUpdate.push({
+        type: 'CREATE',
+        relation: newRel
+      })
+    }
+  })
+
+  if (relatedRelToUpdate.length === 0) {
+    return null
+  }
+
+  return relatedRelToUpdate
+}
+
+/**
+ * Make the new related object converted for the related contact
+ * @param {import('cozy-client/types/types').IOCozyContact} originalContactRelation - Original contact relation
+ * @param {string} originalContactId - Original contact id
+ */
+export const makeRelationMapping = (
+  originalContactRelation,
+  originalContactId
+) => {
+  const relationTypesMapping = getRelationTypesMapping(
+    originalContactRelation.metadata.relationTypes
+  )
+  return {
+    _id: originalContactId,
+    _type: DOCTYPE_CONTACTS,
+    metadata: {
+      relationTypes: relationTypesMapping
+    }
+  }
+}
+
+/**
+ * @param {Object} params
+ * @param {import('cozy-client/types/CozyClient').default} params.client - CozyClient
+ * @param {import('../types').RelatedRelationships} params.relation - Related relationships to update
+ * @param {string} params.originalContactId - Original contact id
+ * @returns {Promise<import('cozy-client/types/types').IOCozyContact[]>} - Related contacts
+ */
+export const createRelatedContact = async ({
+  client,
+  relation,
+  originalContactId
+}) => {
+  try {
+    const contactsQueryById = buildContactsQueryById(relation._id)
+    const { data: relatedContact } = await client.query(
+      contactsQueryById.definition()
+    )
+    const relationConverted = makeRelationMapping(relation, originalContactId)
+    const updatedRelatedContact = setHasManyItem(
+      relatedContact,
+      'related',
+      originalContactId,
+      relationConverted
+    )
+
+    return updatedRelatedContact
+  } catch (error) {
+    log.error('Error creating related contact', error)
+  }
+}
+
+/**
+ * @param {Object} params
+ * @param {import('cozy-client/types/CozyClient').default} params.client - CozyClient
+ * @param {import('../types').RelatedRelationships} params.relation - Related relationships to update
+ * @param {string} params.originalContactId - Original contact id
+ * @returns {Promise<import('cozy-client/types/types').IOCozyContact[]>} - Related contacts
+ */
+export const updateRelatedContact = async ({
+  client,
+  relation,
+  originalContactId
+}) => {
+  try {
+    const contactsQueryById = buildContactsQueryById(relation._id)
+    const { data: relatedContact } = await client.query(
+      contactsQueryById.definition()
+    )
+    const relationConverted = makeRelationMapping(relation, originalContactId)
+    const updatedRelatedContact = setHasManyItem(
+      relatedContact,
+      'related',
+      originalContactId,
+      relationConverted
+    )
+
+    return updatedRelatedContact
+  } catch (error) {
+    log.error('Error updating related contact', error)
+  }
+}
```

Comment:
So `createRelatedContact` and `updateRelatedContact` are equal ? Maybe you can use only one function called `upsertRelatedContact`. But it's your choice, sometimes it's good to have the different path explicit.

Reply:
In fact, I prefer to separate the different actions.
(Initially, I was also thinking of using another CC helper in this function).

Reply:
I will update the code because it wasn't working when removing a “relationTypes”.
This is a problem on the cozy-client side, so I'm updating “manually” to avoid delaying the feature.
See issue https://github.com/cozy/cozy-client/issues/1560